fastly/terraform/main.tf

```diff
@@ -1,65 +1,31 @@
 provider "fastly" {
-  version = "0.11.1"
+  version = "~> 0.27"
 }
 
+locals {
+  vcl = fileset("${path.module}/../vcl", "*.vcl")
+}
 output "service_id" {
   value = ["${fastly_service_v1.app.id}"]
 }
 
 resource "fastly_service_v1" "app" {
-  name = "placeholder"
+  name = "placeholder" // This gets replaced by the name defined in dev_override.tf/qa_override.tf/production_override.tf
 
   force_destroy = false
 
-  vcl {
-    name    = "main.vcl"
-    content = file("${path.module}/../vcl/main.vcl")
-    main    = true
-  }
-
-  vcl {
-    name    = "polyfill-service.vcl"
-    content = file("${path.module}/../vcl/polyfill-service.vcl")
-  }
-
-  vcl {
-    name    = "normalise-user-agent-3-25-1.vcl"
-    content = file("${path.module}/../vcl/normalise-user-agent-3-25-1.vcl")
-  }
-
   vcl {
     name    = "normalise-user-agent.vcl"
     content = file("${path.module}/../../node_modules/@financial-times/polyfill-useragent-normaliser/lib/normalise-user-agent.vcl")
   }
 
-  vcl {
-    name    = "fastly-boilerplate-begin.vcl"
-    content = file("${path.module}/../vcl/fastly-boilerplate-begin.vcl")
-  }
-
-  vcl {
-    name    = "fastly-boilerplate-end.vcl"
-    content = file("${path.module}/../vcl/fastly-boilerplate-end.vcl")
-  }
-
-  vcl {
-    name    = "breadcrumbs.vcl"
-    content = file("${path.module}/../vcl/breadcrumbs.vcl")
-  }
-
-  vcl {
-    name    = "redirects.vcl"
-    content = file("${path.module}/../vcl/redirects.vcl")
-  }
-
-  vcl {
-    name    = "synthetic-responses.vcl"
-    content = file("${path.module}/../vcl/synthetic-responses.vcl")
-  }
-
-  vcl {
-    name    = "top_pops.vcl"
-    content = file("${path.module}/../vcl/top_pops.vcl")
+  dynamic "vcl" {
+    for_each = local.vcl
+    content {
+      name    = vcl.value
+      main    = vcl.value == "main.vcl" ? true : false
+      content = templatefile("${path.module}/vcl/${vcl.value}", {})
```

Comment:
You *might* not need templatefile here, but I guess it's handy!

Reply:
Yes, we don't really need it as we are not providing any variables to the vcl files right now but I figured I'd keep it around to keep it similar to other Fastly configs in the company.